lib/policy/policyValidator.js

```diff
@@ -0,0 +1,117 @@
+'use strict'; // eslint-disable-line strict
+
+const Ajv = require('ajv');
+const userPolicySchema = require('./userPolicySchema');
+const errors = require('../errors');
+
+const ajValidate = new Ajv({ allErrors: true });
+// compiles schema to functions and caches them for all cases
+const userPolicyValidate = ajValidate.compile(userPolicySchema);
+
+const errDict = {
+    required: {
+        Version: 'Policy document must be version 2012-10-17 or greater.',
+        Action: 'Policy statement must contain actions.',
+    },
+    pattern: {
+        Action: 'Actions/Conditions must be prefaced by a vendor,' +
+            ' e.g., iam, sdb, ec2, etc.',
+        Resource: 'Resource must be in ARN format or "*".',
+    },
+    minItems: {
+        Resource: 'Policy statement must contain resources.',
+    },
+};
+
+// parse ajv errors and build list of erros
+function _parseErrors(ajvErrors) {
```

Comment:
True, this function has changed a lot since it was initially written. I have upgraded the description reflecting the current usage.